aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -265,16 +442,25 @@ void max_pool2d_with_indices_backward_out_cuda_template(
   const int dilationH = safe_downcast<int, int64_t>(dilation[0]);
   const int dilationW = dilation.size() == 1 ? dilationH : safe_downcast<int, int64_t>(dilation[1]);
 
-  TORCH_CHECK((input_.ndimension() == 3 || input_.ndimension() == 4),
-    "non-empty 3D or 4D (batch mode) tensor expected for input");
-
-  const Tensor input = input_.contiguous();
+  const auto memory_format = input_.suggest_memory_format();
+  if (memory_format == at::MemoryFormat::ChannelsLast) {
```

Comment:
PLease support channels last 3D tensors

Reply:
we are not able to create 3D Tensor with channels_last memory format:
torch.rand(2, 2, 2).contiguous(memory_format=torch.channels_last)
RuntimeError: required rank 4 tensor to use channels_last format

Reply:
Ok, lets leave it as is now.